internal_coroutines_test.go

```diff
@@ -1034,3 +1034,53 @@ func TestDecodeFutureChain(t *testing.T) {
 	}
 	require.EqualValues(t, expected, history)
 }
+
+func TestSelectFuture_WithBatchSets(t *testing.T) {
+	var history []string
+	d := newDispatcher(background, func(ctx Context) {
+		future1, settable1 := NewFuture(ctx)
+		future2, settable2 := NewFuture(ctx)
+		future3, settable3 := NewFuture(ctx)
+
+		s := NewSelector(ctx)
```

Comment:
But GO doesn't have something equivalent to future to compare behaviours. Here is the play ground example for reading on a channel, it seems to always pick first one. https://play.golang.org/p/cSDH3vxMtH
Our go on select future will also pick first callback always.